### ISSUE TITLE
rust: build fails with llvm-3.9.x

### ISSUE THREAD

**Issue by pullmoll** [voidlinux/void-packages#4782]
If I patch `configure` to accept llvm versions `(3.[7-9]*)` with this sed in `post_extract()`
```
        sed -i configure -e 's;(3\.\[7-8\]\*);(3.[7-9]*);' 
```
the build fails later on with this error:
```
CMake Error at cmake/Modules/AddCompilerRT.cmake:1 (include):
  include could not find load file:

    AddLLVM
Call Stack (most recent call first):
  lib/CMakeLists.txt:4 (include)


-- Configuring incomplete, errors occurred!
See also "/builddir/rustc-1.11.0/build/x86_64-unknown-linux-gnu/compiler-rt/build/CMakeFiles/CMakeOutput.log".
See also "/builddir/rustc-1.11.0/build/x86_64-unknown-linux-gnu/compiler-rt/build/CMakeFiles/CMakeError.log".
thread '<main>' panicked at '
command did not execute successfully, got: exit code: 1

build script failed, must exit now', .cargo/registry/src/github.com-1ecc6299db9ec823/cmake-0.1.17/src/lib.rs:463
note: Run with `RUST_BACKTRACE=1` for a backtrace.
make: *** [Makefile:49: dist] Error 101
=> ERROR: rust-1.11.0_1: failed to run do_build() at line 14.
```

The file `/usr/lib/cmake/llvm/AddLLVM.cmake` is there, though.

**Comment by chneukirchen** [comment]
https://github.com/rust-lang/rust/issues/34867

Not sure we can backport that easily.

**Comment by pullmoll** [comment]
I thought about trying to copy the AddLLVM.cmake file(s) over to where AddCompilerRT.cmake lives, because rust is stopping me from compiling firefox for x86_64 gcc6.

**Comment by pullmoll** [comment]
With `cp -pRv /usr/lib/cmake/llvm/* src/compiler-rt/cmake/Modules/` the (next) error looks like this:
```
-- Configuring incomplete, errors occurred!
See also "/builddir/rustc-1.11.0/build/x86_64-unknown-linux-gnu/compiler-rt/build/CMakeFiles/CMakeOutput.log".
See also "/builddir/rustc-1.11.0/build/x86_64-unknown-linux-gnu/compiler-rt/build/CMakeFiles/CMakeError.log".
thread '<main>' panicked at '
command did not execute successfully, got: exit code: 1

build script failed, must exit now', .cargo/registry/src/github.com-1ecc6299db9ec823/cmake-0.1.17/src/lib.rs:463
note: Run with `RUST_BACKTRACE=1` for a backtrace.
make: *** [Makefile:49: dist] Error 101
=> ERROR: rust-1.11.0_2: failed to run do_build() at line 14.
```

**Comment by chneukirchen** [comment]
Need to scroll above to really see the error.

Try
`sed -i 's,\${LLVM_CMAKE_PATH}/LLVMConfig.cmake,/usr/lib/cmake/llvm/LLVMConfig.cmake,' src/compiler-rt/CMakeLists.txt`

**Comment by pullmoll** [comment]
This doesn't work. I see the same AddLLVM not found error as above.

Looking at the CMakeLists.txt code you patched, I'll try with `-DLLVM_CMAKE_PATH=${XBPS_CROSS_BASE}/usr` in configure_args.

**Comment by pullmoll** [comment]
Still the same... *fsck*

**Comment by chneukirchen** [comment]
Jürgen Buchmüller <notifications@github.com> writes:

> This doesn't work. I see the same AddLLVM not found error as above.
>
> Looking at the CMakeLists.txt code you patched, I'll try with
> `-DLLVM_CMAKE_PATH=${XBPS_CROSS_BASE}/usr` in configure_args.

It built here compiler-rt at least...

You need both approaches.

-- 
Christian Neukirchen  <chneukirchen@gmail.com>  http://chneukirchen.org

**Comment by pullmoll** [comment]
Hey, ho and a bottle of rum. That did the trick!
It compiles with gcc-6.2.0 now as well.
Will you push a patch (hack), or should I?
Now at least firefox will be available for the gcc6 environment.

**Comment by pullmoll** [comment]
I first thought that fixed it (it is better, but not yet everything that is required), because then comes the next error: [rusi build.log](http://sprunge.us/LWXd)

**Comment by chneukirchen** [comment]
Delaying until next Rust release, which should work with 3.9.

**Comment by chneukirchen** [comment]
Fixed in d5ca5ae.
